components/homme/src/share/prim_driver_base.F90

```diff
@@ -474,13 +492,17 @@ subroutine prim_init1_geometry(elem, par, dom_mt)
 
     if (topology=="cube") then
        if(par%masterproc) write(iulog,*) "initializing cube elements..."
+#ifndef HOMME_WITHOUT_PIOLIBRARY
```

Comment:
Some of this code is getting cluttered with cpp stuff. Does it really not compile w/o these? If it _does_ compile and is just a logical/usage error, I recommend erroring with an informative message but removing all the ifdefs. Ideally, the ifdefs would surround just lines of code that won't compile, e.g., the lines in the unstructured mesh module.